client/components/deposits-overview/recent-deposits-list.tsx

```diff
@@ -93,6 +96,38 @@ const RecentDepositsList: React.FC< RecentDepositsProps > = ( {
 		return <></>;
 	}
 
+	// Add a notice indicating the potential business day delay for pending and in_transit deposits.
+	let bannerAdded = false;
+	const depositRows = [ ...deposits ]
+		.reverse() // reverse the array so that the oldest pending or in_transit deposit has the notice added to it.
+		.map( ( deposit ) => {
+			let bannerNotice = null;
+
+			if (
+				! bannerAdded &&
+				( 'pending' === deposit.status ||
+					'in_transit' === deposit.status )
+			) {
+				bannerNotice = (
+					<BannerNotice
+						status="info"
+						icon={ <InfoOutlineIcon /> }
+						children={ strings.notices.businessDayDelay }
+						isDismissible={ false }
+					/>
+				);
+				bannerAdded = true;
+			}
+
+			return (
+				<Fragment key={ deposit.id }>
+					<DepositTableRow deposit={ deposit } />
+					{ bannerNotice }
+				</Fragment>
+			);
+		} )
+		.reverse(); // reverse the array back to the original order.
```

Comment:
To simplify this code, I'd suggest we determine the id of the deposit that needs the banner and provide it to the loop. Just my opinion, feel free to leave this as is, the outcome will be the same.

```suggestion
	// Add a notice indicating the potential business day delay for pending and in_transit deposits.
	const earliestPendingDepositId = [ ...deposits ]
		.reverse()
		.find(
			( deposit ) =>
				'pending' === deposit.status || 'in_transit' === deposit.status
		)?.id;

	const depositRows = deposits.map( ( deposit ) => (
		<Fragment key={ deposit.id }>
			<DepositTableRow deposit={ deposit } />
			{ deposit.id === earliestPendingDepositId && (
				<BannerNotice
					status="info"
					icon={ <InfoOutlineIcon /> }
					children={ strings.notices.businessDayDelay }
					isDismissible={ false }
				/>
			) }
		</Fragment>
	) );
```

Reply:
Love it. This is much cleaner. Applied in [46cad35](https://github.com/Automattic/woocommerce-payments/pull/5839/commits/46cad35dbefdda7f8314faadef50950d743e7e5f).